### PR TITLE
[doc][yba][2024.1.1] Kubernetes Prometheus

### PR DESCRIPTION
**Pull request by ddhodge**
Use a custom Prometheus server with Kubernetes
DOC-411

@netlify /stable/yugabyte-platform/alerts-monitoring/prometheus-custom/

### CODE REVIEW

File: docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+headcontent: Export universe metrics to your own Prometheus instance
+menu:
+  stable_yugabyte-platform:
+    parent: alerts-monitoring
+    identifier: prometheus-setup
+    weight: 90
+type: docs
+---
+
+YugabyteDB Anywhere provides [metrics](../anywhere-metrics/) via a Prometheus instance that is created with your YugabyteDB Anywhere installation.
```
Comment (yushenng) [inline docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md:15]: Slight revision to these lines:

While YugabyteDB Anywhere includes an embedded Prometheus instance and uses it for storing time-series metrics data, you can also set up your own independent and separately-managed Prometheus instance.

This additional Prometheus instance can collect, visualize, alert on, and analyze universe metrics in your own observability tools.   Prometheus itself, or any tool able to utilize Prometheus as a data source (such as Grafana) can be used.

How this works: your separate independent Prometheus instance will scrape data from DB nodes directly (in the case of VM-based universes) or scrape data from a K8s Prometheus Service Monitor (in the case of K8s-based universes).    This data scraping runs in parallel and independently with respect to YBA's own embedded Prometheus.

File: docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+This additional Prometheus instance can collect, visualize, alert on, and analyze universe metrics in your own observability tools. Prometheus itself, or any tool able to use Prometheus as a data source (such as Grafana) can be used.
+
+Your independent Prometheus instance will scrape data from database nodes directly (in the case of VM-based universes) or scrape data from a K8s Prometheus Service Monitor (in the case of K8s-based universes). This data scraping runs in parallel to and independently from the YugabyteDB Anywhere-embedded Prometheus.
+
+## Kubernetes
+
+YugabyteDB allows scraping of universe metrics in Kubernetes using a ServiceMonitor. Similar to a Pod or Deployment, ServiceMonitor is also a resource that is handled by the Prometheus Operator. The ServiceMonitor resource selects a Kubernetes Service based on a given set of labels. With the help of this selected Service, the Prometheus Operator creates scrape configurations for Prometheus. Prometheus uses these configurations to actually scrape the application pods.
```
Comment (amannijhawan) [inline docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md:23]: This is not required, 
we can point them to prometheus operator in details.

File: docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+The ServiceMonitor resource can have details like port names or numbers, scrape interval, HTTP path where application is exposing the metrics, and so on. In the case of the `yugabytedb/yugabyte` Helm chart, it creates ServiceMonitors for YB-Master as well as YB-TServer pods.
+
+### Prerequisites
+
+Install the Prometheus Operator on your Kubernetes universe as follows:
```
Comment (amannijhawan) [inline docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md:29]: We also don't need to give directions to install prometheus operator, this is something every customer might do differently, these instructions are also not correct.

File: docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        release: prom
+    ```
+
+    This configuration enables the service monitor and adds an extra label `release: prom`, which is required by the Prometheus Operator to discover this service monitor.
+
+1. Click **Upgrade**.
+
+To verify the configuration:
+
+1. Access your Prometheus dashboard.
```
Comment (amannijhawan) [inline docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-setup.md:63]: Screenshots are missing from the docs

File: docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-custom/prometheus-scrape.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+1. Click **Upgrade**.
+
+To verify the configuration:
+
+1. Access your Prometheus dashboard.
+
+1. Navigate to the **Service Discovery** page and search for the name of your universe.
+
+    The service monitor for your YugabyteDB universe should be listed.
```
Comment (aishwarya24) [inline docs/content/stable/yugabyte-platform/alerts-monitoring/prometheus-custom/prometheus-scrape.md:87]:
```suggestion
    The service monitor for your YugabyteDB Anywhere universe should be listed.
```

**Review by aishwarya24 — APPROVED** [review]

**Pull request merged**